Examples/Example-iOS/AppDelegate.swift

```diff
@@ -11,35 +12,44 @@ func throwingFunc() throws {
 
 @UIApplicationMain
 final class AppDelegate: UIResponder, UIApplicationDelegate {
-
-    let fileUrl = createAndWriteFile("sample.txt")
-
+    
     var window: UIWindow?
     
     func application(_ application: UIApplication,
                      didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        let backtraceCredentials = BacktraceCredentials(endpoint: URL(string: Keys.backtraceUrl as String)!,
-                                                        token: Keys.backtraceToken as String)
+        
+        // Usage https://docs.saucelabs.com/error-reporting/platform-integrations/ios/configuration/#usage
+        let backtraceCredentials = BacktraceCredentials(endpoint: URL(string: Keys.backtraceUrl as String)!, token: Keys.backtraceToken as String)
 
+        // Customize Database Settings https://docs.saucelabs.com/error-reporting/platform-integrations/ios/configuration/#database-settings
         let backtraceDatabaseSettings = BacktraceDatabaseSettings()
         backtraceDatabaseSettings.maxRecordCount = 10
+        
+        // BacktraceClientConfiguration https://docs.saucelabs.com/error-reporting/platform-integrations/ios/configuration/#advanced-usage
         let backtraceConfiguration = BacktraceClientConfiguration(credentials: backtraceCredentials,
                                                                   dbSettings: backtraceDatabaseSettings,
                                                                   reportsPerMin: 10,
                                                                   allowsAttachingDebugger: true,
                                                                   detectOOM: true)
-        BacktraceClient.shared = try? BacktraceClient(configuration: backtraceConfiguration)
-        BacktraceClient.shared?.attributes = ["foo": "bar", "testing": true]
-        BacktraceClient.shared?.attachments.append(fileUrl)
-        BacktraceClient.shared?.delegate = self
         
-        do {
-            try throwingFunc()
-        } catch {
-            BacktraceClient.shared?.send(attachmentPaths: []) { (result) in
-                print("AppDelegate:Result:\(result)")
-            }
+        // Customize PLCrashReporterConfig with custom basePath https://docs.saucelabs.com/error-reporting/platform-integrations/ios/configuration/#plcrashreporter
+        guard let plcrashReporterConfig = PLCrashReporterConfig(signalHandlerType: .BSD, symbolicationStrategy: .all, basePath: crashDirectory.path) else {
```

Comment:
what if we move to Crashpad? The dedicated settings object for it should be in the database options. This provides an abstraction layer on top of the crash reporting solution. We should stick to the database options - I'm against introducing "plcrashreporter specific" settings.